### PR TITLE
Broken links tools.ietf.org/html/

### LINKED ISSUE #1750: Broken links tools.ietf.org/html/
In code there are still links provided in comments that point to tools.ietf.org/html/ resources

These links redirect to datatracker.ietf.org/doc/html and either fail to redirect or redirect. Providing the link to datatracker.ietf.org would minimize the risk of the failure to redirect happening.

### PR DESCRIPTION
**Pull request by cesarpayan**
changing tools.ietf.org/html/ to datatracker.ietf.org/doc/html
Also changing some http to https in these links

### CODE REVIEW

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs [hunk 1]
```diff
@@ -28,69 +28,69 @@
...
 {
     /// <summary>
-    /// List of header parameter names see: http://tools.ietf.org/html/rfc7519#section-5.
+    /// List of header parameter names see: https://datatracker.ietf.org/doc/html/rfc7519#section-5.
     /// </summary>
     public struct JwtHeaderParameterNames
     {
         /// <summary>
-        /// see:https://tools.ietf.org/html/rfc7515#section-4.1.1
+        /// see:https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.1
```
Comment (ciaozhang) [inline src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs]: nit: See: https, we probably need a space here

File: src/Microsoft.IdentityModel.Tokens/SecurityAlgorithms.cs [hunk 2]
```diff
@@ -108,7 +108,7 @@ public static class SecurityAlgorithms
         public const string RsaSsaPssSha384 = "PS384";
         public const string RsaSsaPssSha512 = "PS512";
 
-        // See : https://tools.ietf.org/html/rfc7518#section-5.1
+        // See : https://datatracker.ietf.org/doc/html/rfc7518#section-5.1
```
Comment (ciaozhang) [inline src/Microsoft.IdentityModel.Tokens/SecurityAlgorithms.cs]: nit: extra space

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs (hunk 1 above)
Comment (ciaozhang) [reply src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs]: also other change in this file as well

File: src/Microsoft.IdentityModel.Tokens/SecurityAlgorithms.cs [hunk 3]
```diff
@@ -63,37 +63,37 @@ public static class SecurityAlgorithms
...
         public const string Sha512Digest = "http://www.w3.org/2001/04/xmlenc#sha512";
 
-        // See: https://tools.ietf.org/html/rfc7518#appendix-A
+        // See: https://datatracker.ietf.org/doc/html/rfc7518#appendix-A
         public const string Sha256 = "SHA256";
         public const string Sha384 = "SHA384";
         public const string Sha512 = "SHA512";
 
-        // See https://tools.ietf.org/html/rfc6931#section-2.3.6
+        // See https://datatracker.ietf.org/doc/html/rfc6931#section-2.3.6
```
Comment (ciaozhang) [inline src/Microsoft.IdentityModel.Tokens/SecurityAlgorithms.cs]: nit missing : 
also missing :  someplace in this file

File: src/System.IdentityModel.Tokens.Jwt/JwtHeaderParameterNames.cs [hunk 4]
```diff
@@ -28,69 +28,69 @@
...
 {
     /// <summary>
-    /// List of header parameter names see: http://tools.ietf.org/html/rfc7519#section-5.
+    /// List of header parameter names see: https://datatracker.ietf.org/doc/html/rfc7519#section-5.
     /// </summary>
     public struct JwtHeaderParameterNames
     {
         /// <summary>
-        /// see:https://tools.ietf.org/html/rfc7515#section-4.1.1
+        /// see:https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.1
```
Comment (ciaozhang) [inline src/System.IdentityModel.Tokens.Jwt/JwtHeaderParameterNames.cs]: nit: add a space after See: 
also other changes in this file as well

**Review by brentschmaltz — APPROVED** [review]

**Review by brentschmaltz — CHANGES_REQUESTED** [review]
There are some spacing issue and : and casing.

File: src/Microsoft.IdentityModel.Tokens/SecurityAlgorithms.cs (hunk 2 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Tokens/SecurityAlgorithms.cs]: It is good to be consistent with spacing and casing.

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs (hunk 1 above)
Comment (cesarpayan) [reply src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs]: fixed in commit adbc4d9b31ee1f48b70e12644e833dc636aa9089

File: src/Microsoft.IdentityModel.Tokens/SecurityAlgorithms.cs (hunk 2 above)
Comment (cesarpayan) [reply src/Microsoft.IdentityModel.Tokens/SecurityAlgorithms.cs]: fixed in commit adbc4d9b31ee1f48b70e12644e833dc636aa9089

File: src/Microsoft.IdentityModel.Tokens/SecurityAlgorithms.cs (hunk 3 above)
Comment (cesarpayan) [reply src/Microsoft.IdentityModel.Tokens/SecurityAlgorithms.cs]: fixed in commit adbc4d9b31ee1f48b70e12644e833dc636aa9089

File: src/System.IdentityModel.Tokens.Jwt/JwtHeaderParameterNames.cs (hunk 4 above)
Comment (cesarpayan) [reply src/System.IdentityModel.Tokens.Jwt/JwtHeaderParameterNames.cs]: fixed in commit adbc4d9b31ee1f48b70e12644e833dc636aa9089

**Review by ciaozhang — APPROVED** [review]

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs [hunk 5]
```diff
@@ -28,69 +28,69 @@
...
-        /// see:https://tools.ietf.org/html/rfc7515#section-4.1.1
+        /// See: https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.1
         /// </summary>
         public const string Alg = "alg";
 
         /// <summary>
-        /// see:https://tools.ietf.org/html/rfc7515#section-4.1.10
-        /// also:https://tools.ietf.org/html/rfc7519#section-5.2
+        /// See: https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.10
+        /// also:https://datatracker.ietf.org/doc/html/rfc7519#section-5.2
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs]:
```suggestion
        /// Also: https://datatracker.ietf.org/doc/html/rfc7519#section-5.2
```

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs [hunk 6]
```diff
@@ -28,69 +28,69 @@
...
 
         /// <summary>
-        /// see:https://tools.ietf.org/html/rfc7518#section-4.7.1.1
+        /// See: https://datatracker.ietf.org/doc/html/rfc7518#section-4.7.1.1
         /// </summary>
         public const string IV = "iv";
 
         /// <summary>
-        /// see:https://tools.ietf.org/html/rfc7515#section-4.1.2
+        /// see:https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.2
```
Comment (jennyf19) [inline src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs]:
```suggestion
        /// See: https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.2
```

File: src/System.IdentityModel.Tokens.Jwt/JwtHeaderParameterNames.cs [hunk 7]
```diff
@@ -28,69 +28,69 @@
...
-        /// see:https://tools.ietf.org/html/rfc7515#section-4.1.1
+        /// See: https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.1
         /// </summary>
         public const string Alg = Microsoft.IdentityModel.JsonWebTokens.JwtHeaderParameterNames.Alg;
 
         /// <summary>
-        /// see:https://tools.ietf.org/html/rfc7515#section-4.1.10
-        /// also:https://tools.ietf.org/html/rfc7519#section-5.2
+        /// See: https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.10
+        /// also: https://datatracker.ietf.org/doc/html/rfc7519#section-5.2
```
Comment (jennyf19) [inline src/System.IdentityModel.Tokens.Jwt/JwtHeaderParameterNames.cs]:
```suggestion
        /// Also: https://datatracker.ietf.org/doc/html/rfc7519#section-5.2
```

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs [hunk 8]
```diff
@@ -28,69 +28,69 @@
...
-        /// see:https://tools.ietf.org/html/rfc7515#section-4.1.4
+        /// See: https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.4
         /// </summary>
         public const string Kid = "kid";
 
         /// <summary>
-        /// see:https://tools.ietf.org/html/rfc7515#section-4.1.9
-        /// also:https://tools.ietf.org/html/rfc7519#section-5.1
+        /// See: https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.9
+        /// also:https://datatracker.ietf.org/doc/html/rfc7519#section-5.1
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs]: For consistency, also -> Also

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs (hunk 5 above)
Comment (cesarpayan) [reply src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs]: changed in e89d4fbafb9ac32deb2b8b8a704e473b9e647669

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs (hunk 6 above)
Comment (cesarpayan) [reply src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs]: Thanks. Missed that one, good catch. Fixed in e89d4fbafb9ac32deb2b8b8a704e473b9e647669

File: src/System.IdentityModel.Tokens.Jwt/JwtHeaderParameterNames.cs (hunk 7 above)
Comment (cesarpayan) [reply src/System.IdentityModel.Tokens.Jwt/JwtHeaderParameterNames.cs]: Changed in e89d4fbafb9ac32deb2b8b8a704e473b9e647669

File: src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs (hunk 8 above)
Comment (cesarpayan) [reply src/Microsoft.IdentityModel.JsonWebTokens/JwtHeaderParameterNames.cs]: changed in e89d4fbafb9ac32deb2b8b8a704e473b9e647669

**Review by brentschmaltz — APPROVED** [review]

**Pull request merged**